src/plugins/data_source_management/public/components/create_data_source_wizard/components/create_form/create_data_source_form.tsx

```diff
@@ -55,8 +59,12 @@ export interface CreateDataSourceState {
   description: string;
   endpoint: string;
   auth: {
-    type: AuthType;
-    credentials: UsernamePasswordTypedContent | SigV4Content | undefined;
+    type: AuthType | string;
+    credentials:
+      | UsernamePasswordTypedContent
+      | SigV4Content
+      | { [key: string]: string }
```

Comment:
nit: should we extract to interface?

Reply:
Thanks for the comment~ 
I will prefer not using a interface here since registered auth credentials should not be a specific type.